### PR TITLE
Python script transform - necessary ui changes

### PR DESCRIPTION
**Pull request by anwar6953**
This PR includes the necessary changes to configure a python transform from the UI.
The definition of the python transform is external to this repository.

http://builds.cask.co/browse/CDAP-RBT515-2

### CODE REVIEW

**Comment by awholegunch** [comment]
Python -> Jython ? Now is the time to do it...

File: cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java [hunk 1]
```diff
@@ -188,4 +188,26 @@ private Object convertString(Schema schema, String strVal) throws UnexpectedForm
...
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    StructuredRecord that = (StructuredRecord) o;
+
+    return schema.equals(that.schema) && fields.equals(that.fields);
```
Comment (albertshau) [inline cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java]: nit: to be consistent with other places, can use Objects.equals()

File: cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java [hunk 2]
```diff
@@ -188,4 +188,26 @@ private Object convertString(Schema schema, String strVal) throws UnexpectedForm
...
+
+  }
+
+  @Override
+  public int hashCode() {
+    int result = schema.hashCode();
```
Comment (albertshau) [inline cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java]: nit: to be consistent with other places, can use Objects.hash()

**Comment by anwar6953** [comment]
@awholegunch  Well, the changes in this PR only include a new widget - editor for python syntax.
It's not actually having anything to do with the plugin that uses that editor.

File: cdap-ui/templates/common/Script.json [hunk 3]
```diff
@@ -7,7 +7,6 @@
        "fields" : {
           "script" : {
              "widget": "javascript-editor",
-             "label" : "Script",
```
Comment (awholegunch) [inline cdap-ui/templates/common/Script.json]: Why did you want to remove the label? Should it not be "Javascript"?

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/sink/TimePartitionedFileSetDatasetAvroSink.java [hunk 4]
```diff
@@ -94,8 +94,6 @@ public void initialize(BatchRuntimeContext context) throws Exception {
   public void transform(StructuredRecord input,
                         Emitter<KeyValue<AvroKey<GenericRecord>, NullWritable>> emitter) throws Exception {
     emitter.emit(new KeyValue<>(new AvroKey<>(recordTransformer.transform(input)), NullWritable.get()));
-    InvalidEntry entry = new InvalidEntry(100, "dummy", input);
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/sink/TimePartitionedFileSetDatasetAvroSink.java]: ogosh when was this added?

**Comment by albertshau** [comment]
small comments, otherwise lgtm

File: cdap-ui/templates/common/Script.json (hunk 3 above)
Comment (anwar6953) [inline cdap-ui/templates/common/Script.json]: I recall @ajainarayanan mentioning that its not necessary.
Can you clarify, @ajainarayanan ?

**Comment by ajainarayanan** [comment]
:+1: LGTM on UI changes.

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/sink/TimePartitionedFileSetDatasetAvroSink.java (hunk 4 above)
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/sink/TimePartitionedFileSetDatasetAvroSink.java]: I think @gsps1 accidentally put it in, when he was testing emitting errors.

File: cdap-ui/templates/common/Script.json (hunk 3 above)
Comment (awholegunch) [inline cdap-ui/templates/common/Script.json]: If you remove it, I don't think you'll be able to set the label of that node.

**Comment by awholegunch** [comment]
After looking at the code, I change my mind and think this should stay as "python", as it is pulling in the Python editor, which is what is being used for the Jython plugin. So, that makes sense. Who knows, maybe we'll have a Jython-aware editor one day...

File: cdap-ui/templates/common/Script.json (hunk 3 above)
Comment (anwar6953) [inline cdap-ui/templates/common/Script.json]: ok, I reverted the change.

**Comment by albertshau** [comment]
tiny comment, lgtm

**Comment by awholegunch** [comment]
LGTM

**Comment by albertshau** [comment]
lgtm

**Comment by ajainarayanan** [comment]
LGTM

**Comment by anwar6953** [comment]
Starting build again with new changes. Will merge pending build.

**Pull request merged**